File: 420-E71 (Maintenance Project)/CSAdmin/CSAdmin-2018-MR-Team2/CSAdminCode/Common/Format.vb
Namespace Common
    Public Class Format
        ''' <summary>
        ''' Method that formats date. eg. Date: 2011-05-14 Formatted: 2011-MAY-14
        ''' </summary>
        ''' <param name="EvalDate">Date to be formatted</param>
        ''' <returns>Formatted date</returns>
        ''' <remarks>Author: Renee Ghattas</remarks>
        Public Shared Function FormatDate(ByVal EvalDate As Date) As String
            Dim theResult As String = String.Empty

            theResult = EvalDate.ToString("yyyy-MMM-dd")

            Return theResult
        End Function

        ''' <summary>
        ''' Method that formats year and semester. eg. YearSemester: 20121 Formatted: Winter 2012
        ''' </summary>
        ''' <param name="YearSemester">Year and Semester to be formatted</param>
        ''' <returns>Formatted year and semester</returns>
        ''' <remarks>Author: Renee Ghattas</remarks>
        Public Shared Function FormatSemester(ByVal YearSemester As Integer) As String
            Dim theResult As String = String.Empty
            Dim snum As Integer = CType(Right(YearSemester, 1), Integer)
            Dim y As String = YearSemester.ToString.Substring(0, YearSemester.ToString.Length - 1)

            If snum = 0 Then
                theResult = Nothing
            ElseIf snum = 1 Then
                theResult = "Winter " & y
            ElseIf snum = 2 Then
                theResult = "Summer " & y
            ElseIf snum = 3 Then
                theResult = "Fall " & y
            End If

            Return theResult
        End Function

        ''' <summary>
        ''' Method that formats first and last name. eg. FirstName: Renee LastName: Ghattas Formatted: Ghattas, Renee
        ''' </summary>
        ''' <param name="FirstName">First name to be formatted</param>
        ''' <param name="LastName">Last name to be formatted</param>
        ''' <returns>Formatted full name</returns>
        ''' <remarks>Author: Renee Ghattas</remarks>
        Public Shared Function FormatLastFirst(ByVal FirstName As String, ByVal LastName As String) As String
            Dim theResult As String = String.Empty

            theResult = LastName & ", " & FirstName

            Return theResult
        End Function

        ''' <summary>
        ''' Method that formats first and last name. eg. FirstName: Renee LastName: Ghattas Formatted: Renee Ghattas
        ''' </summary>
        ''' <param name="FirstName">First name to be formatted</param>
        ''' <param name="LastName">Last name to be formatted</param>
        ''' <returns>Formatted full name</returns>
        ''' <remarks>Author: Renee Ghattas</remarks>
        Public Shared Function FormatFirstLast(ByVal FirstName As String, ByVal LastName As String) As String
            Dim theResult As String = String.Empty

            theResult = FirstName & " " & LastName

            Return theResult
        End Function
    End Class
End Namespace

File: 420-E71 (Maintenance Project)/CSAdmin-2018-MR-Team1/CSAdmin-2018-MR-Team1/CSAdmin2/Logic/Format.cs
using System;

namespace CSAdmin2.Logic
{
	public class Format
	{
		public static string FormatDate(DateTime EvalDate)
		{
			//stop reserving memory for no reason.
			return EvalDate.ToString("yyyy-MMM-dd");
		}

		/// <summary>
		///     Method that formats year and semester. eg. YearSemester: 20121 Formatted: Winter 2012
		/// </summary>
		/// <param name="YearSemester"></param>
		/// <returns></returns>
		public static string FormatSemester(short YearSemester)
		{
			//holy sweet potatoes this needed some optimisation and clean up - GUI
			string res;
			switch (YearSemester % 10)
			{
				case 1:
					res = $"Winter ";
					break;
				case 2:
					res = $"Summer ";
					break;
				case 3:
					res = $"Fall ";
					break;
				default:
					return null;
			}

			//because integer division is awesome
			res += (YearSemester / 10).ToString();

			return res;
		}

		/// <summary>
		///     Method that formats first and last name. eg. FirstName: Renee LastName: Ghattas Formatted: Ghattas, Renee
		/// </summary>
		/// <param name="FirstName"></param>
		/// <param name="LastName"></param>
		/// <returns></returns>
		public static string FormatLastFirst(string FirstName, string LastName)
		{
			//because interpolated strings are badass - GUI
			return $"{LastName}, {FirstName}";
		}

		/// <summary>
		///     Method that formats first and last name. eg. FirstName: Renee LastName: Ghattas Formatted: Renee Ghattas
		/// </summary>
		/// <param name="FirstName"></param>
		/// <param name="LastName"></param>
		/// <returns></returns>
		public static string FormatFirstLast(string FirstName, string LastName)
		{
			//same reason as before - GUI
			return $"{FirstName} {LastName}";
		}
	}
}